### PR TITLE
chore(django): improve snapshot tests

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The non-asgi snapshots previously used the pytest-django app to generate
snapshots. This somewhat worked but had the drawbacks of not knowing
what pytest-django did during startup and teardown (like doing database
clean up and tear down) which could generate spans and cause the
snapshots to fail.

The solution is to move the non-asgi snapshots to use real django
applications spawned in subprocesses (like the asgi snapshots). This
should help make the tests less flaky as well as give us the flexibility
in the future to more easily add new django apps.

https://pymotw.com/2/subprocess/#process-groups-sessions was a handy resource.

### CODE REVIEW

File: tests/contrib/django/test_django_snapshots.py [hunk 1]
```diff
@@ -1,226 +1,277 @@
...
 
     try:
         yield client
     finally:
         resp = client.get_ignored("/shutdown-tracer")
-        assert resp.status_code == 200
+        assert resp.status_code == 200, resp.text
+        import time
+
+        time.sleep(0.5)
```
Comment (Kyle-Verhoog) [inline tests/contrib/django/test_django_snapshots.py:110]: there is a race here between the test, app and test agent.

At this point the app has submitted the traces to the test agent but there isn't a guarantee that the test agent has finished processing them by the time the test asks the test agent to perform the snapshot test.

I'll add a comment.

**Comment by Kyle-Verhoog** [comment]
I've re-run the Django job a few times now and all 🟢 so 🤞 this addresses the flakiness!

**Comment by majorgreys** [comment]
@Kyle-Verhoog If you re-enable the middleware spans now, do you find the flakiness returns?

**Comment by majorgreys** [comment]
Should we expect this to change much the total duration?

**Review by P403n1x87 — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
😩 still some flakiness present 

```
==================================== ERRORS ====================================
________________ ERROR at teardown of test_urlpatterns_include _________________
At request <Request GET /test/session/snapshot >:
   At snapshot (token='tests.contrib.django.test_django_snapshots.test_urlpatterns_include_21x'):
    - Directory: /snapshots
    - CI mode: 1
    - Trace File: /snapshots/tests.contrib.django.test_django_snapshots.test_urlpatterns_include_21x.json
    - Stats File: /snapshots/tests.contrib.django.test_django_snapshots.test_urlpatterns_include_21x_tracestats.json
    At compare of 1 expected trace(s) to 0 received trace(s):
Did not receive expected traces: 'django.request'
```

**Review by ZStriker19 — APPROVED** [review]

**Pull request closed without merging**